Source/Details/ASMainSerialQueue.mm

```diff
@@ -40,14 +40,12 @@ - (NSUInteger)numberOfScheduledBlocks
 
 - (void)performBlockOnMainThread:(dispatch_block_t)block
 {
-
-  AS::UniqueLock l(_serialQueueLock);
-  [_blocks addObject:block];
   {
-    l.unlock();
-    [self runBlocks];
-    l.lock();
+    AS::UniqueLock l(_serialQueueLock);
```

Comment:
You don't need a `AS::UniqueLock` here anymore. Let's just use `AS::MutexLocker`

Reply:
Done.